quickfort.lua

```diff
@@ -87,5 +100,94 @@ files are described in the files themselves.
 .. _blueprints/library: https://github.com/DFHack/dfhack/tree/develop/data/blueprints/library
 .. _original Quickfort documentation: https://github.com/joelpt/quickfort#manual-material-selection
 ]====]
+end
+
+local settings = {
+    blueprints_dir = "blueprints",
+    force_marker_mode = false,
+    force_interactive_build = false,
+}
+
+do_set = function(args)
+    if #args == 0 then
+        print("active settings:")
+        printall(settings)
+        return
+    end
+    if #args ~= 2 then
+        error('error: expected "quickfort set [<key> <value>]"')
+    end
+    if settings[args[1]] == nil then
+        error('error: invalid setting: "' .. args[1] .. '"')
+    end
+    val = args[2]
+    if type(settings[args[1]]) == "boolean" then
+        val = args[2] == "true"
+    end
+    settings[args[1]] = val
+    print("successfully set " .. args[1] .. " to '" .. tostring(val) .. "'")
+end
+
+local utils = require('utils')
+
+local valid_list_args = utils.invert({
+    'l',
+    '-library',
+})
+
+do_list = function(in_args)
+    local args = utils.processArgs(in_args, valid_list_args)
+    local show_library = args['l'] ~= nil or args['-library'] ~= nil
+    print("NOT YET IMPLEMENTED")
+    print(string.format(
+        "would call list with show_library=%s", tostring(show_library)))
+end
+
+local valid_commands = utils.invert({
+    'run',
+    'orders',
+    'undo',
+})
+
+local valid_command_args = utils.invert({
+    'q',
+    '-quiet',
+    'v',
+    '-verbose',
+    's',
+    '-sheet',
+})
+
+do_command = function(action, in_args)
+    if not valid_commands[action] then
+        error(string.format('invalid command: "%s"', action))
+    end
+
+    local args = utils.processArgs(in_args, valid_command_args)
+    local quiet = args['q'] ~= nil or args['-quiet'] ~= nil
+    local verbose = args['v'] ~= nil or args['-verbose'] ~= nil
+    local sheet = args['s'] ~= nil or args['-sheet'] ~= nil
+
+    print("NOT YET IMPLEMENTED")
+    print(string.format(
+        "would call %s with quiet=%s, verbose=%s, sheet=%s",
+        tostring(quiet), tostring(verbose), sheet))
+end
+
+end -- if not initialized
+
+-- main
+local argv = {...}
+local action = table.remove(argv, 1)
 
-print('Given arguments: ',...)
+if action == 'set' then
```

Comment:
Not sure what is consensus on big if/elseif (i.e. switch statements) but another way to do this is a table with functions
```lua
switch={ list=do_list, run=do_command, orders=do_command}
switch[action](argv)
```

Reply:
done. thanks!